x/stake/keeper/delegation.go

```diff
@@ -795,3 +825,69 @@ func (k Keeper) ValidateUnbondAmount(
 
 	return shares, nil
 }
+
+func (k Keeper) crossDistributeUndelegated(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress, amount int64) (sdk.Events, sdk.Error) {
+	relayFeeCalc := fees.GetCalculator(types.CrossDistributeUndelegatedRelayFee)
+	if relayFeeCalc == nil {
+		return sdk.Events{}, sdk.ErrInternal("no fee calculator of distributeUndelegated")
+	}
+	relayFee := relayFeeCalc(nil)
+	bscRelayFee := bsc.ConvertBCAmountToBSCAmount(relayFee.Tokens.AmountOf(k.BondDenom(ctx)))
+	bscTransferAmount := new(big.Int).Sub(bsc.ConvertBCAmountToBSCAmount(amount), bscRelayFee)
```

Comment:
Please do not transfer undelegated BNB back to BSC when it is not enough to cover the relayerFee.

Reply:
Done